workers/social/lib/social/models/computeproviders/stacktemplate.coffee

```diff
@@ -160,19 +173,72 @@ module.exports = class JStackTemplate extends Module
     providersParser    = clientRequire 'app/lib/util/stacks/providersparser'
     requirementsParser = clientRequire 'app/lib/util/stacks/requirementsparser'
 
+    config.groupStack       ?= no
     config.requiredData      = requirementsParser template
     config.requiredProviders = providersParser template, supportedProviders
 
     return config
 
 
-  validateTemplate = (template, group, callback) ->
+  generateSlugFromTitle = ({ originId, group, title, index }, callback) ->
+
+    unless title
+      return callback null
+
+    slug = _title = if index? then "#{title} #{index}" else title
+    slug = slugify slug
+
+    JStackTemplate.count { originId, group, slug }, (err, count) ->
+      return callback err  if err?
+
+      if count is 0
+        callback null, { slug, title: _title }
+      else
+        index ?= 0
+        index += 1
+        generateSlugFromTitle { originId, group, title, index }, callback
+
 
+  validateTemplateData = (data, options, callback) ->
+
+    unless data
+      return callback new KodingError 'Template data is required!'
+
+    { config = {}, title, template, rawContent, templateDetails } = data
+
+    if template and rawContent
+      template = generateTemplateObject template, rawContent, templateDetails
+      if config
+        config = updateConfigForTemplate config, template.rawContent
+        title ?= generateTemplateTitle config.requiredProviders[0]
+
+    { originId, group } = options
     limitConfig = helpers.getLimitConfig group
-    return callback null  unless limitConfig.limit # No limit, no pain.
+    group       = group.slug
+
+    generateSlugFromTitle { group, originId, title }, (err, res = {}) ->
+
+      return callback err  if err
+
+      { slug, title } = res
+      replacements = { template, config, title, slug }
+
+      if not data.template     or \
+         not limitConfig.limit or \
+             limitConfig.limit is 'unlimited' # No limit, no pain.
+        return callback null, replacements
+
+      ComputeProvider = require './computeprovider'
+      ComputeProvider.validateTemplateContent \
+        data.template, limitConfig, (err) -> callback err, replacements
```

Comment:
lets add it if we need it, also it is against our flow to return both error and the possible partial response - in most cases